lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -703,20 +703,21 @@ push @tests, __PACKAGE__->new(
 
 # There's no AP uniqueness requirement?
 push @tests, __PACKAGE__->new(
-    test_query => "select *
-                     from ap
-                    where invnumber in (select invnumber
-                                               from ap
-                                              group by invnumber
-                                              having count(*) > 1)
-                    order by invnumber",
-    display_name => $locale->text('Non-unique invoice numbers'),
+    test_query => "SELECT id, concat(invnumber,'-',row_number() over(partition by invnumber order by id)) AS invnumber,
+                          transdate, duedate, datepaid, ordnumber, quonumber, approved
+                     FROM ap
+                    WHERE invnumber IN (SELECT invnumber
+                                          FROM ap
+                                      GROUP BY invnumber
+                                        HAVING count(*) > 1)
+                 ORDER BY invnumber",
+    display_name => $locale->text('Non-unique invoice numbers detected'),
     name => 'no_duplicate_ap_invoicenumbers',
     display_cols => ['id', 'invnumber', 'transdate', 'duedate', 'datepaid',
                      'ordnumber', 'quonumber', 'approved'],
     column => 'invnumber',
  instructions => $locale->text(
-                   'Please make all AP invoice numbers unique'),
+                   'Please review suggestions to make all AP invoice numbers unique'),
```

Comment:
Maybe we should list the conflicting data as well as the suggested data and explain something about that in the text above?